openid-connect-server/src/main/java/org/mitre/oauth2/web/DeviceEndpoint.java

```diff
@@ -139,6 +139,7 @@ public String requestDeviceCode(@RequestParam("client_id") String clientId, @Req
 			response.put("device_code", dc.getDeviceCode());
 			response.put("user_code", dc.getUserCode());
 			response.put("verification_uri", config.getIssuer() + USER_URL);
+			response.put("verification_uri_complete", config.getIssuer() + USER_URL + "?user_code=" + dc.getUserCode());
```

Comment:
It makes me nervous not using a URL Builder for this. Yes, I know that the user code we're generating is already URL safe but I'd rather have it be explicitly handled in case someone replaces the user code generator in a custom service.